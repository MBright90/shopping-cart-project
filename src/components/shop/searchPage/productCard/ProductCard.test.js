import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'

import ProductCard from './index'

describe('ProductCard', () => {
  let props
  beforeEach((props = { product: { name: 'exampleName', image: 'exampleImage', price: '999' } }))

  it('renders to match snapshot', () => {
    const { container } = render(<ProductCard {...props} />)
    expect(container).toMatchSnapshot()
  })

  it('sets the correct image source', () => {
    render(<ProductCard {...props} />)

    const productImage = screen.getAllRole('img')
    expect(productImage).toHaveStyle('background-image: url("exampleImage")')
  })

  it('calls handleProductClick when clicked', () => {
    const handleProductClick = jest.fn()
    render(<ProductCard {...props} />)

    const productCard = screen.getByTestId('product-card')
    userEvent.click(productCard)
    expect(handleProductClick).toHaveBeenCalled()
  })

  it('calls handleProductKeyDown when clicked', () => {
    const handleProductKeyDown = jest.fn()
    render(<ProductCard {...props} />)

    const productCard = screen.getByTestId('product-card')
    userEvent.keyDown(productCard, 'Enter')
    expect(handleProductKeyDown).toHaveBeenCalled()
  })
})
