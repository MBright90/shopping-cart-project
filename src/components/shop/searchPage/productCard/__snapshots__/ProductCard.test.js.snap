// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders to match snapshot 1`] = `
<div>
  <a
    class="productCard"
    data-testid="product-card"
    href="#/products/undefined"
    tabindex="-1"
  >
    <div
      class="productImage"
      role="img"
      style="background-image: url(exampleImage);"
    />
    <div
      class="detailsContainer"
    >
      <p
        class="name"
      >
        exampleName
      </p>
      <p
        class="price"
      >
        Â£
        999
      </p>
    </div>
  </a>
</div>
`;
